towhee/runtime/performance_profiler.py

```diff
@@ -0,0 +1,245 @@
+                                          'ts': n_tracer['queue_in'][0]*1000000, 'cat': n_tracer['name'] + '_' + n_id})
+                else:
+                    profiler_json.append({'ph': 'b', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': n_tracer['name'],
+                                          'ts': n_tracer['init_in'][0]*1000000, 'cat': n_tracer['name'] + '_' + n_id})
+                profiler_json.append({'ph': 'e', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': n_tracer['name'],
+                                      'ts': n_tracer['queue_out'][0]*1000000, 'cat': n_tracer['name'] + '_' + n_id})
+
+                if len(n_tracer['init_in']) != 0:
+                    profiler_json.append({'ph': 'B', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'init',
+                                          'ts': n_tracer['init_in'][0]*1000000})
+                    profiler_json.append({'ph': 'E', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'init',
+                                          'ts': n_tracer['init_out'][0]*1000000})
+                profiler_json.append({'ph': 'B', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'wait_data',
+                                      'ts': n_tracer['queue_in'][0]*1000000})
+                profiler_json.append({'ph': 'E', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'wait_data',
+                                      'ts': n_tracer['process_in'][0]*1000000})
+                for p_in, p_out in zip(n_tracer['process_in'], n_tracer['process_out']):
+                    profiler_json.append({'ph': 'B', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'call_op',
+                                          'ts': p_in*1000000})
+                    profiler_json.append({'ph': 'E', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'call_op',
+                                          'ts': p_out*1000000})
+                profiler_json.append({'ph': 'B', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'output_data',
+                                      'ts': n_tracer['process_out'][-1]*1000000})
+                profiler_json.append({'ph': 'E', 'pid': pipe_name, 'tid': n_tracer['name'], 'id': n_id, 'name': 'output_data',
+                                      'ts': n_tracer['queue_out'][0]*1000000})
+        return profiler_json
+
+
+class Event:
+    pipe_name = '_run_pipe'
+    pipe_in = 'pipe_in'
+    pipe_out = 'pipe_out'
+    init_in = 'init_in'
+    init_out = 'init_out'
+    process_in = 'process_in'
+    process_out = 'process_out'
+    queue_in = 'queue_in'
+    queue_out = 'queue_out'
+
+
+class TimeProfiler:
+    """
+    TimeProfiler to record the timestamp.
+    """
+    def __init__(self):
+        self._enable = False
+        self.time_record = []
+
+    def record(self, name, event, pipe_input=None):
```

Comment:
input & output nodes also need tracer.

Reply:
Get it, and do we need to show the performance for them?

Reply:
yeah